### ISSUE TITLE
clarification on using prebuilt cross-compiled OTP needed

### ISSUE THREAD

**Issue by dch** [grisp/rebar3_grisp#41]
I'm assuming a great deal from this blog post[1] that if I have e.g. OTP20.x from
my system package manager, then I should be able to produce a grisp release
without needing to build RTEMS from scratch, if:

- there is no C dependency
- my OTP release matches whatever you have in "the CDN"

I've tried with the 2 latest OTP releases (OTP 21, OTP 20) and haven't had
great success, using the robot example with included wifi support:

```
$ erl +V
Erlang (SMP,ASYNC_THREADS,HIPE) (BEAM) emulator version 9.3.3.3
$ rebar3 do compile, grisp deploy -n robot -v 0.1.0
===> Verifying dependencies...
===> Compiling robot
===> [{relname,"robot"},{relvsn,"0.1.0"},{force,false}]
===> Current Erlang version ("20") does not match target Erlang version (undefined). It is not guaranteed that the deployed release will work!
===> Verifying dependencies...
===> Compiling robot
===> Starting relx build process ...
===> Resolving OTP Applications from directories:
          /repos/grisp/sdcard/app/_build/grisp/lib
          /repos/grisp/sdcard/app/_grisp/otp/21.0/install
===> Failed to solve release:
 Dependency kernel is specified as a dependency but is not reachable by the system.
```

some questions:

- what supported versions do you have in "the CDN"? is there a way to see it?
- how can I include my own matching version from my RTEMS source build so it matches whatever my desktop package manager is currently using?

```
> uname -a
FreeBSD akai.skunkwerks.at 12.0-ALPHA9 FreeBSD 12.0-ALPHA9 r339356+891833d897fd(master) GENERIC  amd64
> pkg info erlang-runtime20
erlang-runtime20-20.3.8.9
Name           : erlang-runtime20
Version        : 20.3.8.9
Installed on   : Thu Oct  4 14:07:43 2018 UTC
Origin         : lang/erlang-runtime20
Architecture   : FreeBSD:12:amd64
Prefix         : /usr/local
Categories     : java lang parallel
Licenses       : APACHE20
Maintainer     : erlang@FreeBSD.org
WWW            : http://www.erlang.org/
Comment        : Functional programming language from Ericsson
Options        :
	DIRTY          : on
	DTRACE         : on
	GS             : off
	HIPE           : on
	JAVA           : off
	KQUEUE         : on
	NATIVE         : off
	ODBC           : off
	OPENSSL        : on
	SCTP           : on
	THREADS        : on
	WX             : off
Annotations    :
	FreeBSD_version: 1200084
	repo_type      : binary
	repository     : FreeBSD
Flat size      : 103MiB
Description    :
Erlang is a programming language used to build massively scalable soft
real-time systems with requirements on high availability. Some of its
uses are in telecoms, banking, e-commerce, computer telephony and
instant messaging. Erlang's runtime system has built-in support for
concurrency, distribution and fault tolerance.

This port contains a standalone runtime environment of Erlang 20
to be used during the development of OTP applications.

WWW: http://www.erlang.org/
```

[1]: https://mailchi.mp/stritzinger/prebuilt-cross-otp-or-how-to-get-started-while-on-a-train

**Comment by eproxus** [comment]
@dch Can you post your `rebar.config`? It looks like you don't have the target GRiSP Erlang version configured there... (if so, we should really have a better error for that)

* Currently there's no way to see which Erlang version are on the CDN, unfortunately
* If you build Erlang yourself, you need the same version on your system _and_ cross-compiled. However this should be done automatically if you have the `{build, ...}` section configured in the GRiSP configuration in `rebar.config`

**Comment by eproxus** [comment]
This should now be solved in [version 2.2.0](https://github.com/grisp/rebar3_grisp/releases/tag/2.2.0) by running `rebar3 grisp package list` to see which pre-built versions exist. In addition, selecting a partial version (like `"23"`) will automatically select the latest available package when deploying.
